libafl_qemu/Cargo.toml

```diff
@@ -145,9 +145,9 @@ bindgen = "0.69"
 cc = "1.0"
 
 [dev-dependencies]
-static_assertions = "1.1"
-nix = { version = "0.29", features = ["signal", "process"] }
-proc-maps = "0.3"
+static_assertions = "1.1.0"
+nix = { version = "0.29.0", features = ["signal", "process"] }
```

Comment:
why do we want to specify the sub-version, too?

Reply:
In general I think it is better to specify the full version number.
If I'm developing on version 1.2.3 by specifying the full version I don't have to check that it also works on 1.2.2, 1.2.1 ecc.
Also, from Cargo docs:
> Although it looks like a specific version of the crate, it actually specifies a range of versions and allows SemVer compatible updates

Reply:
i do not agree here. [From semver](https://semver.org/#spec-item-6): PATCHes are used for bugfixes that do not break compatibility (cargo doc also agrees on this point).
if crates follow the convention, we should never need to check for retrocompatibility on PATCH bumps.
i don't think we will ever want to keep a buggier version of a dependency that is still fully compatible, except if the dependency authors break the conversion.
if we really want to have the exact versions, i'd say it's the exact purpose of lock files no?

Reply:
> i don't think we will ever want to keep a buggier version of a dependency that is still fully compatible

we wont do this, as mentioned:

> Although it looks like a specific version of the crate, it actually specifies a range of versions and allows SemVer compatible updates

Reply:
It's a quite subtle thing but let's say we are using dep foo 1.2.3 that is perfectly API compatible with 1.2.2, but 1.2.2 has a bug (that we never experienced since on newer version). By putting 1.2.3 anyone compiling the library will use **at least** 1.2.3 and not 1.2.2.

Reply:
Np! :)
It should be an extremely rare case, but I still don't see any reason not to include it

Reply:
So far we haven't done it and it worked well, no issues as far as I'm aware of

Reply:
I think we should go with `x.y` as per semantic versioning, unless if we run into issues with a specific one

Reply:
> Np! :) It should be an extremely rare case, but I still don't see any reason not to include it

The reason is that we'll have outdated dependencies and it'll be harder to mix and match libafl with other libraries relying on the same depdenencies with a different sub version

Reply:
> The reason is that we'll have more outdated dependencies and it'll be harder to mix and match libafl with other libraries relying on the same depdenencies with a different sub version

Mmm, nope 😅

TLDR of the previous messages: with `foo="1.2.3"` cargo is allowed to use any semver compatible update, so >=`1.2.3` and <`2.0.0`

I'd say this doesn't restrict compatibility in 99.99% of the cases with respect to `foo="1.2"`
That remaining 0.01% might happen when in another crate we want to use there is some crazy stuff like `foo="=1.2.2"` (note the `=`) or `foo=">= 1.2.0, <1.2.3"`

Reply:
Ohhh. WHAT :D
Ok in that case ...